fill/fill_station.cc

```diff
@@ -70,14 +74,27 @@ class CommanderServiceImpl final : public Commander::Service
         if (request->qd_retract()) {
              qd.Actuate();
         }
+
+        if (request->bv1_open()){
+            ball_valve.open();
+        } else {
```

Comment:
You need an outer if to check request->has_bv1_open() first. Right now this else will close the valve any time any other command is issued

Reply:
Can you explain further what you mean?

Reply:
The commands in the request proto are all optional which mean they will only exist in the request when a command related to it is commanded. If we retract the QD for example, the ball valve commands will not be in request. With your current code, that would tell the ball valve to close since request->bv1_open() would not be true. You'll need to check if the request has the relevant field before checking the value. Like:
if (request->has_bv1_open()) {
    if (request->bv1_open()) {
        // open bv
    } else {
        // close bv
    }
}